src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,21 +1,77 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
-@Controller
+@RestController
+@RequestMapping("/books")
 public class BookController {
 
-    @GetMapping("/greeting")
-    public String greeting(
-        @RequestParam(name = "name", required = false, defaultValue = "Mundo") String name,
-        Model model) {
+    private final BookRepository bookRepository;
 
-        model.addAttribute("name", name);
+    public BookController(BookRepository bookRepository) {
 
-        return "greeting";
+        this.bookRepository = bookRepository;
+
+    }
+
+    @GetMapping()
+    public Iterable<Book> findAll() {
+
+        return bookRepository.findAll();
+
+    }
+
+    @GetMapping("/{id}")
+    public Book findById(@PathVariable Long id) {
+
+        return bookRepository.findById(id)
+            .orElseThrow(BookNotFoundException::new);
+
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book save(@RequestBody Book book) {
+
+        return bookRepository.save(book);
+
+    }
+
+    @PutMapping("/{id}")
+    public Book update(@RequestBody Book book, @PathVariable Long id) {
+
+        if (book.getId() != id) {
+            throw new ResponseStatusException(HttpStatus.PRECONDITION_FAILED,
```

Comment:
You may add another exception to handle this case, the same thing you've done for the BookNotFoundException.